ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp

```diff
@@ -0,0 +1,724 @@
+                return;
+            for (auto &&poFeature : poStopsLyr)
+            {
+                const char *pszStopId = poFeature->GetFieldAsString(nStopIdIdx);
+                if (pszStopId)
+                {
+                    m_oMapStopIdToLonLat[pszStopId] = std::make_pair(
+                        poFeature->GetFieldAsDouble(nStopLonIdx),
+                        poFeature->GetFieldAsDouble(nStopLatIdx));
+                }
+            }
+        }
+
+        auto poStopTimesDS = std::unique_ptr<GDALDataset>(GDALDataset::Open(
+            (m_osDirname + "/stop_times.txt").c_str(), GDAL_OF_VECTOR));
+        if (!poStopTimesDS)
+            return;
+        auto poStopTimesLyr = poStopTimesDS->GetLayer(0);
+        if (!poStopTimesLyr)
+            return;
+        const auto poStopTimesLyrDefn = poStopTimesLyr->GetLayerDefn();
+        const int nStopIdIdx = poStopTimesLyrDefn->GetFieldIndex("stop_id");
+        const int nTripIdIdx = poStopTimesLyrDefn->GetFieldIndex("trip_id");
+        const int nStopSequenceIdx =
+            poStopTimesLyrDefn->GetFieldIndex("stop_sequence");
+        if (nStopIdIdx < 0 || nTripIdIdx < 0 || nStopSequenceIdx < 0)
+            return;
+        for (auto &&poFeature : poStopTimesLyr)
+        {
+            const char *pszStopId = poFeature->GetFieldAsString(nStopIdIdx);
+            const char *pszTripId = poFeature->GetFieldAsString(nTripIdIdx);
+            const int nStopSequence =
+                poFeature->GetFieldAsInteger(nStopSequenceIdx);
+            if (pszStopId && pszTripId)
+            {
+                m_oMapTripIdToStopIds[pszTripId][nStopSequence] = pszStopId;
+            }
+        }
+    }
+    catch (const std::exception &)
+    {
+        CPLError(CE_Failure, CPLE_OutOfMemory, "Not enough memory");
+    }
+}
+
+/***********************************************************************/
+/*                          GetNextFeature()                           */
+/***********************************************************************/
+
+OGRFeature *OGRGTFSLayer::GetNextFeature()
+{
+    if (m_bIsTrips && !m_bPrepared)
+        PrepareTripsData();
+
+    while (true)
+    {
+        auto poSrcFeature =
+            std::unique_ptr<OGRFeature>(m_poUnderlyingLayer->GetNextFeature());
+        if (poSrcFeature == nullptr)
+            return nullptr;
+
+        auto poFeature = cpl::make_unique<OGRFeature>(m_poFeatureDefn);
+        const int nFieldCount = poSrcFeature->GetFieldCount();
+        poFeature->SetFID(poSrcFeature->GetFID());
+        auto poSrcLayerDefn = m_poUnderlyingLayer->GetLayerDefn();
+        for (int i = 0; i < nFieldCount; ++i)
+        {
+            const auto eType = m_poFeatureDefn->GetFieldDefn(i)->GetType();
+            if (poSrcLayerDefn->GetFieldDefn(i)->GetType() == eType)
+            {
+                poFeature->SetField(i, poSrcFeature->GetRawFieldRef(i));
+            }
+            else if (eType == OFTDate)
```

Comment:
Maybe a switch? I see only date and int are actually handled and default to string.

Reply:
A switch construct isn't the most natural one here given that the first ``if`` doesn't test eType against a constant but against an expression